x-pack/legacy/plugins/siem/public/containers/detection_engine/rules/api.ts

```diff
@@ -26,19 +27,16 @@ import { DETECTION_ENGINE_RULES_URL } from '../../../../common/constants';
  * @param signal to cancel request
  */
 export const addRule = async ({ rule, signal }: AddRulesProps): Promise<NewRule> => {
-  const response = await fetch(`${chrome.getBasePath()}${DETECTION_ENGINE_RULES_URL}`, {
+  const response = await npStart.core.http.fetch<NewRule>(DETECTION_ENGINE_RULES_URL, {
     method: rule.id != null ? 'PUT' : 'POST',
     credentials: 'same-origin',
-    headers: {
-      'content-type': 'application/json',
-      'kbn-xsrf': 'true',
-    },
     body: JSON.stringify(rule),
+    asResponse: true,
```

Comment:
@FrankHassanabad -- should these DE API requests have the `'kbn-system-api': 'true'` header?

Reply:
Yeah, I just noticed we're not super consistent with our headers on these. Should we be sending:
* credentials: same-origin
* kbn-system-api: true

For all these requests?

Reply:
Followup: 
* `credentials: same-origin` is the default and doesn't need to be specified (4d92808). 
* `kbn-system-api: true` is used to determine whether or not a request will extend the client's session (if it's a system call, it will not)
    * I'm not aware of the default session TTL for kibana, but given the 'run in background' nature of detections I'm not sure what the answer should be, here. This almost feels like a product question.

@spong @FrankHassanabad what do you think?

Reply:
Per slack conversations I am good with all of this.